spec/factories/custom_button.rb

```diff
@@ -2,5 +2,10 @@
   factory :custom_button do
     sequence(:name)        { |n| "custom_button_#{seq_padded_for_sorting(n)}" }
     sequence(:description) { |n| "custom_button_#{seq_padded_for_sorting(n)}" }
+
+    trait :with_resource_action_dialog do
+      name             { "button1" }
```

Comment:
Why do you need a name at all?  It will inherit the sequence from the parent...then when you create it just use the name given to you.